subprojects/core-api/src/main/java/org/gradle/api/toolchain/management/ToolchainManagementSpec.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.toolchain.management;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.plugins.ExtensionAware;
+
+/**
+ * Configures how toolchains are resolved. Is extended at runtime to support both
+ * currently existing toolchains (Java) and future ones.
+ * <p>
+ * One option available at present is using the <code>jvm-toolchains</code> plugin, which
+ * adds a <code>jdks</code> block to it. This block allows for specifying which
+ * <code>JavaToolchainRepository</code>s to use and in what order, if java toolchain
+ * auto-provisioning is needed.
+ *
+ * @since 7.6
+ */
+@Incubating
+public interface ToolchainManagementSpec extends ExtensionAware {
```

Comment:
`Spec` in Gradle is a term usually linked to what the JDK names `Predicate` or our extension of this - a configured object that is supposed to map something.
I know that Settings has the `PluginManagementSpec` but it feels wrong as well.

So I would name this type `ToolchainManagement` (similar to the `DependencyManagement` type) or `ToolchainManagementHandler`